rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/compile_time_interpreter.kt

```diff
@@ -139,36 +139,64 @@ open class BaseCompileTimeInterpreter(
             val field = it.fields.find { it.name.equals(declName, ignoreCase = true) }
             if (field != null) return (field.elementSize() /*/ field.declaredArrayInLine!!*/)
         }
-        rContext.statement()
-                .forEach {
-                    when {
-                        it.dspec() != null -> {
-                            val name = it.dspec().ds_name().text
-                            if (name == declName) {
-                                // TODO verify...
-                                return it.dspec().TO_POSITION().text.asInt()
+        return findSize(rContext.statement(), declName, conf, false)!!
```

Comment:
null assertions operation only if really needed, and considering the implementation of `findSize` for which  returns always `null` when innerBlock parameter is `true,` this approach is too much error prone.

Reply:
[Read this comment](https://github.com/smeup/jariko/pull/422#discussion_r1500460016) first